bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/services/WorkbenchServiceRegistry.java

```diff
@@ -84,34 +85,34 @@ public boolean hasService(Class<?> api) {
 		}
 	};
 
-	private Map factories = new HashMap();
+	private final Map<String, ServiceFactoryHandle> factories = new HashMap<>();
 
 	static class ServiceFactoryHandle {
 		AbstractServiceFactory factory;
-		WeakHashMap serviceLocators = new WeakHashMap();
```

Comment:
was there a good reason to use WeakHashMap instead of a Set?

Reply:
i don't know. As the Name says it is weak -i.e. it can be GCed. probably on purpose

Reply:
And now it can't be GCed any more. Is that an issue?

Reply:
why could'nt it? its still a WeakHashMap under the hood

Reply:
Ah. Sure. I didn't see that. Sorry for the confusion.